dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaYarnApplicationsTask.java

```diff
@@ -0,0 +1,181 @@
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.dto.ApiYARNApplicationDTO;
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.dto.ApiYARNApplicationListDTO;
+import com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager.dto.ApiYARNApplicationTypeDTO;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.IOException;
+import java.io.Writer;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** The task dumps YARN applications from Cloudera Manager API */
+public class ClouderaYarnApplicationsTask extends AbstractClouderaManagerTask {
+  private static final Logger LOG = LoggerFactory.getLogger(ClouderaYarnApplicationsTask.class);
+  private static final DateTimeFormatter isoDateTimeFormatter =
+      DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+
+  private final String[] predefinedAppTypes = {
+    "MAPREDUCE",
+  };
+  private final int includedLastDays;
+
+  public ClouderaYarnApplicationsTask(int days) {
+    super(String.format("yarn-applications-%d.jsonl", days));
+    this.includedLastDays = days;
+  }
+
+  @Override
+  protected void doRun(
+      TaskRunContext context, @Nonnull ByteSink sink, @Nonnull ClouderaManagerHandle handle)
+      throws Exception {
+    List<ClouderaClusterDTO> clusters = handle.getClusters();
+    Preconditions.checkNotNull(
+        clusters, "Clusters must be initialized before fetching YARN applications.");
+
+    List<ApiYARNApplicationDTO> totalYARNApplications = new LinkedList<>();
+    YarnApplicationsLoader appLoader =
+        new YarnApplicationsLoader(handle.getApiURI().toString(), handle.getHttpClient());
+    for (ClouderaClusterDTO cluster : handle.getClusters()) {
+      String clusterName = cluster.getName();
+      List<String> yarnAppTypes = fetchYARNApplicationTypes(handle, clusterName);
+      for (String yarnAppType : yarnAppTypes) {
+        LOG.info(
+            String.format(
+                "Dump YARN applications with %s type from %s cluster", yarnAppType, clusterName));
+        List<ApiYARNApplicationDTO> yarnApplications = appLoader.load(clusterName, yarnAppType);
+        totalYARNApplications.addAll(yarnApplications);
+        LOG.info(
+            String.format(
```

Comment:
The same here. Please use logger framework.

Reply:
Thanks, updated.